src/app-layer-parser.c

```diff
@@ -919,9 +919,30 @@ void AppLayerParserTransactionsCleanup(Flow *f, const uint8_t pkt_dir)
     SCLogDebug("start min %"PRIu64, min);
     bool skipped = false;
     // const bool support_files = AppLayerParserSupportsFiles(f->proto, f->alproto);
+    uint32_t tofree_nb = 0;
+    bool free_all = false;
+#define TX_LIST_STEP 0x100
+    if (max_id - min > UINT32_MAX - TX_LIST_STEP) {
```

Comment:
does this mean we're willing to allocate a 4b (-256) array of size 8, so 32GiB? Per thread. Seems we should limit to something much more reasonable, like 1024 or 4096 and fall back to just iterating all txs if exceeded?

Reply:
> Seems we should limit to something much more reasonable, like 1024 or 4096

Yes, but the problem is what to do if we go beyond...

> fall back to just iterating all txs if exceeded?

I am not sure what you mean by "just iterating all txs" ?
This function is meant to clean and free transactions, not just iterate...
In the past, you did not want a generic limit to free all transactions here, because these transactions would not have been properly logged...

In next PR, I will propose a new way : just free 1024 transactions maximum at each run